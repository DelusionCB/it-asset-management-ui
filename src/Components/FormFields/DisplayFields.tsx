import React from 'react';
import {Container, Row} from 'reactstrap';
import {
    extendedApplicationDataProps, extendedDirectoryDataProps,
    LicenseDataProps,
} from '../../Types/types.directories';

interface appProps {
    type: 'application'
    item: extendedApplicationDataProps
}

interface dirProps {
    type: 'directory'
    item: extendedDirectoryDataProps
}

interface lcnProps {
    type: 'license'
    item: LicenseDataProps
}

type combinedProps = appProps | dirProps | lcnProps

function DisplayFields ({type, item}: combinedProps): JSX.Element {
    console.log(item, type, 'item')

    function getFields (): JSX.Element {
        switch (type) {
            case 'application':
                return (
                    <Container className='directoryItem-page'>
                        <Row>
                            <h1>Sovellus</h1>
                        </Row>
                        <Row>
                            <hr />
                            <h2>Sovelluksen kuvaus</h2>
                            <hr />
                        </Row>
                        <Row>
                            <h3>Nimi:</h3>
                            <p>{item.name}</p>
                        </Row>
                        <Row>
                            <h3>Kuvaus:</h3>
                            <p>{item.description}</p>
                        </Row>
                        <Row>
                            <h3>Luokittelu:</h3>
                            <p>{item.classification}</p>
                        </Row>
                        <Row>
                            <h3>Asiakkuus:</h3>
                            <p>{item.place_of_use}</p>
                        </Row>
                        <Row>
                            <h3>Sisältääkö järjestelmä henkilörekisterin:</h3>
                            <p>{item.person_register ? 'Kyllä' : 'Ei'}</p>
                        </Row>
                        <Row>
                            <h3>Henkilötietojen katselun lokitus:</h3>
                            <p>{item.personal_info_logging ? 'Kyllä' : 'Ei'}</p>
                        </Row>
                        <Row>
                            <h3>Asennuksen lisätiedot:</h3>
                            <p>{item.install_info}</p>
                        </Row>
                        <Row>
                            <h3>Avainsanat:</h3>
                            <p>{item.keywords}</p>
                        </Row>
                        <Row>
                            <hr />
                            <h2>Sovelluksen sidokset</h2>
                            <hr />
                        </Row>
                        <Row>
                            <h3>Asennettu palvelimelle:</h3>
                            {item.installed_server
                                ? <>
                                    <p>{item.installed_server.name}</p>
                                </>
                                : <>
                                    <p>Ei asennettu</p>
                                </>
                            }
                        </Row>
                        <Row>
                            <h3>Liittyy palveluihin:</h3>
                            {item.service_dependency
                                ? item.service_dependency.map((child, key) => (
                                    <div key={key}>
                                        <Row>
                                            <h4>Nimi {key + 1}:</h4>
                                            <p>{child.name}</p>
                                        </Row>
                                        <Row>
                                            <h4>Kuvaus {key + 1}:</h4>
                                            <p>{child.description}</p>
                                        </Row>
                                    </div>
                                ))
                                : <>
                                    <h4>Ei palvelukytköksiä</h4>
                                </>
                            }
                        </Row>
                        <Row>
                            <h3>Integraatiot:</h3>
                            {item.integration
                                ? item.integration.map((child, key) => (
                                    <div key={key}>
                                        <Row>
                                            <h4>Nimi {key + 1}:</h4>
                                            <p>{child.name}</p>
                                        </Row>
                                        <Row>
                                            <h4>Kuvaus {key + 1}:</h4>
                                            <p>{child.description}</p>
                                        </Row>
                                    </div>
                                ))
                                : <>
                                    <h4>Ei integraatioita</h4>
                                </>
                            }
                        </Row>
                        <Row>
                            <h3>Lisensointi:</h3>
                            {item.license
                                ? <>
                                    <p>{item.license.name}</p>
                                    <p>{item.license.valid_from_date}</p> - <p>{item.license.valid_until_date}</p>
                                </>
                                : <>
                                    <p>Ei lisenssiä</p>
                                </>
                            }
                        </Row>
                        <Row>
                            <h3>Riippuvaisuudet sovelluksiin / sovelluksista:</h3>
                            {item.application_dependency
                                ? <>
                                    <p>{item.application_dependency.name}</p>
                                </>
                                : <>
                                    <p>Ei riippuvaisuuksia</p>
                                </>
                            }
                        </Row>
                        <Row>
                            <hr />
                            <h2>Jatkuvuus, kriittisyys & palautumistiedot</h2>
                            <hr />
                        </Row>
                        <Row>
                            <h3>Päivitykset:</h3>
                            <p>{item.update_practice}</p>
                        </Row>
                        <Row>
                            <h3>Varmistuskäytännöt & valvonta:</h3>
                            <p>{item.security_practice_monitoring}</p>
                        </Row>
                        <Row>
                            <h3>Palautuskäytännöt & toipuminen:</h3>
                            <p>{item.recovery_practices_convalescence}</p>
                        </Row>
                        <Row>
                            <h3>Käyttöoikeuksien hallinta:</h3>
                            <p>{item.user_rights_management}</p>
                        </Row>
                        <Row>
                            <h3>Tietoturvaratkaisut:</h3>
                            <p>{item.security_solutions}</p>
                        </Row>
                        <Row>
                            <hr />
                            <h2>Vastuut</h2>
                            <hr />
                        </Row>
                        <Row>
                            <h3>Sovelluksen omistaja:</h3>
                            <p>{item.product_owner}</p>
                        </Row>
                        <Row>
                            <h3>Sovelluksen haltija tai muu ensisijainen vastuuhenkilö:</h3>
                            <p>{item.application_holder}</p>
                        </Row>
                        <Row>
                            <h3>Pääkäyttäjät (Tekniset pääkäyttäjät):</h3>
                            <p>{item.admin_users}</p>
                        </Row>
                        <Row>
                            <h3>Vastuu & ammattikäyttäjät (Toiminnalliset pääkäyttäjät, tai esim. Pegasos vastuukäyttäjät):</h3>
                            <p>{item.liability_professional_users}</p>
                        </Row>
                        <Row>
                            <h3>Lisätietoja vastuuhenkilöistä:</h3>
                            <p>{item.holder_extra_info}</p>
                        </Row>
                        <Row>
                            <h3>Käyttöoikeusvastaavat:</h3>
                            <p></p>
                        </Row>
                        <Row>
                            <h3>Toimittaja:</h3>
                            <p>{item.provider}</p>
                        </Row>
                        <Row>
                            <h3>Toimittajan vastuut:</h3>
                            <p>{item.provider_responsibility}</p>
                        </Row>
                        <Row>
                            <h3>Lisäkontaktit:</h3>
                            <p>{item.additional_contacts}</p>
                        </Row>
                        <Row>
                            <hr />
                            <h2>Lisätiedot & linkit</h2>
                            <hr />
                        </Row>
                        <Row>
                            <h3>Lisätietoja:</h3>
                            <p>{item.known_issues}</p>
                        </Row>
                        <Row>
                            <h3>Linkit & dokumentit:</h3>
                            <a>{item.fileUrl}</a>
                        </Row>
                    </Container>
                )
            case 'directory':
                return (
                    <Container>
                        <Row>
                            <h1>Kansio</h1>
                        </Row>
                        <Row>
                            <hr />
                            <h2>Kansion kuvaus</h2>
                            <hr />
                        </Row>
                        <Row>
                            <h3>Nimi:</h3>
                            <p>{item.name}</p>
                        </Row>
                        <Row>
                            <h3>Kuvaus:</h3>
                            <p>{item.description}</p>
                        </Row>
                        <Row>
                            <h3>Sovellukset:</h3>
                            {item.applications.length
                                ? item.applications.map((child, key) => (
                                    <div key={key}>
                                        <Row>
                                            <h4>Nimi {key + 1}:</h4>
                                            <p>{child.name}</p>
                                        </Row>
                                        <Row>
                                            <h4>Kuvaus {key + 1}:</h4>
                                            <p>{child.description}</p>
                                        </Row>
                                    </div>
                                ))
                                : <>
                                    <h4>Ei sovelluksia</h4>
                                </>
                            }
                        </Row>
                        <Row>
                            <h3>Palvelut:</h3>
                            {item.services.length
                                ? item.services.map((child, key) => (
                                    <div key={key}>
                                        <Row>
                                            <h4>Nimi {key + 1}:</h4>
                                            <p>{child.name}</p>
                                        </Row>
                                        <Row>
                                            <h4>Kuvaus {key + 1}:</h4>
                                            <p>{child.description}</p>
                                        </Row>
                                    </div>
                                ))
                                : <>
                                    <h4>Ei sovelluksia</h4>
                                </>
                            }
                        </Row>
                        <Row>
                            <h3>Palvelimet:</h3>
                            {item.servers.length
                                ? item.servers.map((child, key) => (
                                    <div key={key}>
                                        <Row>
                                            <h4>Nimi {key + 1}:</h4>
                                            <p>{child.name}</p>
                                        </Row>
                                        <Row>
                                            <h4>Kuvaus {key + 1}:</h4>
                                            <p>{child.description}</p>
                                        </Row>
                                    </div>
                                ))
                                : <>
                                    <h4>Ei sovelluksia</h4>
                                </>
                            }
                        </Row>
                    </Container>
                )
            case 'license':
                return (
                    <Container>
                        <Row>
                            <h1>Lisenssi</h1>
                        </Row>
                        <Row>
                            <hr />
                            <h2>Lisenssin kuvaus</h2>
                            <hr />
                        </Row>
                        <Row>
                            <h3>Nimi:</h3>
                            <p>{item.name}</p>
                        </Row>
                        <Row>
                            <h3>Kuvaus:</h3>
                            <p>{item.description}</p>
                        </Row>
                        <Row>
                            <h3>Voimassa alkaen:</h3>
                            <p>{item.valid_from_date}</p>
                        </Row>
                        <Row>
                            <h3>Voimassa päättyen:</h3>
                            <p>{item.valid_until_date}</p>
                        </Row>
                        <Row>
                            <h3>Lisenssityyppi:</h3>
                            <p>{item.license_type}</p>
                        </Row>
                        <Row>
                            <h3>Sopimus:</h3>
                            <p>{item.contract}</p>
                        </Row>
                        <Row>
                            <h3>Dokumentit:</h3>
                            <p>{item.fileUrl}</p>
                        </Row>
                    </Container>
                )
        }
    }
    return (
        <div>
            {getFields()}
        </div>
    );
}

export default DisplayFields;
