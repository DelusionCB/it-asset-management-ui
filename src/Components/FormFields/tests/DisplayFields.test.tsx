import React from 'react';
import * as TestRenderer from 'react-test-renderer';
import DisplayFields from '../DisplayFields';
import {BrowserRouter} from 'react-router-dom'
import '@testing-library/jest-dom'
import {
    extendedApplicationDataProps,
} from '../../../Types/types.directories';

describe('<DisplayFields />', () => {
    let renderer: TestRenderer.ReactTestRenderer;
    // Render the component before tests

    const item: extendedApplicationDataProps = {
        base_id: '123',
        name: 'Application Name',
        visibility: 'Public',
        description: 'Application Description',
        created_time: '2023-01-01',
        last_modified_time: '2023-05-01',
        classification: 'Application Classification',
        customership: [],
        application_status: 'Active',
        person_register: true,
        personal_info_logging: false,
        install_info: 'Installation Info',
        keywords: 'Keyword 1, Keyword 2',
        update_practice: 'Update Practice',
        security_practice_monitoring: 'Security Practice Monitoring',
        recovery_practices_convalescence: 'Recovery Practices Convalescence',
        log_archives: 'Log Archives',
        user_rights_management: 'User Rights Management',
        security_solutions: 'Security Solutions',
        product_owner: 'Product Owner',
        application_holder: 'Application Holder',
        admin_users: 'Admin Users',
        liability_professional_users: 'Liability Professional Users',
        holder_extra_info: 'Holder Extra Info',
        provider: null,
        provider_responsibility: 'Provider Responsibility',
        additional_contacts: 'Additional Contacts',
        known_issues: 'Known Issues',
        contract: null,
        fileUrl: null,
        license: null,
        installed_server: [],
        service_dependency: [],
        integration: [],
        application_dependency: [],
    };

    beforeAll(() => {
        renderer = TestRenderer.create(
            <BrowserRouter>
                <DisplayFields<extendedApplicationDataProps>
                    type="application"
                    loading={false}
                    item={item}
                />
            </BrowserRouter>,
        );
    });

    describe('Snapshot', () => {
        it('renders correctly', () => {
            expect(renderer.toJSON()).toMatchSnapshot();
        })
    })
})
