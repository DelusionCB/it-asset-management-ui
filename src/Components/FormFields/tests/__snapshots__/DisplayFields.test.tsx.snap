// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DisplayFields /> Snapshot renders correctly 1`] = `
<div
  className="display-fields"
>
  <div
    className="directory-wrapper container"
  >
    <div
      className="display-header row"
    >
      <h1>
        application.app
      </h1>
    </div>
    <div
      className="display-description row"
    >
      <h2>
        application.description
      </h2>
    </div>
    <div
      className="display row"
    >
      <div
        className="row"
      >
        <h3>
          application.values.name
        </h3>
        <p>
          Application Name
        </p>
      </div>
    </div>
    <div
      className="display row"
    >
      <div
        className="row"
      >
        <h3>
          application.values.description
        </h3>
        <p>
          Application Description
        </p>
      </div>
    </div>
    <div
      className="display row"
    >
      <div
        className="row"
      >
        <h3>
          application.values.classification
        </h3>
        <p>
          Application Classification
        </p>
      </div>
    </div>
    <div
      className="display row"
    >
      <h3>
        application.values.customership
        :
      </h3>
      <h4>
        no-connections
      </h4>
    </div>
    <div
      className="display row"
    >
      <div
        className="row"
      >
        <h3>
          application.values.application_status
        </h3>
        <p>
          select.Active
        </p>
      </div>
    </div>
    <div
      className="display row"
    >
      <div
        className="row"
      >
        <h3>
          application.values.person_register
          :
        </h3>
        <p>
          yes
        </p>
      </div>
    </div>
    <div
      className="display row"
    >
      <div
        className="row"
      >
        <h3>
          application.values.personal_info_logging
          :
        </h3>
        <p>
          no
        </p>
      </div>
    </div>
    <div
      className="display row"
    >
      <div
        className="row"
      >
        <h3>
          application.values.install_info
        </h3>
        <p>
          Installation Info
        </p>
      </div>
    </div>
    <div
      className="display row"
    >
      <div
        className="row"
      >
        <h3>
          application.values.keywords
        </h3>
        <p>
          Keyword 1, Keyword 2
        </p>
      </div>
    </div>
    <div
      className="display-description row"
    >
      <h2>
        application.dependency
      </h2>
    </div>
    <div
      className="display row"
    >
      <h3>
        application.values.installed_server
        :
      </h3>
      <h4>
        no-connections
      </h4>
    </div>
    <div
      className="display row"
    >
      <h3>
        application.values.service_dependency
        :
      </h3>
      <h4>
        no-connections
      </h4>
    </div>
    <div
      className="display row"
    >
      <h3>
        application.values.integration
        :
      </h3>
      <h4>
        no-connections
      </h4>
    </div>
    <div
      className="display row"
    >
      <h3>
        application.values.license
        :
      </h3>
      <div>
        <h4>
          no-connections
        </h4>
      </div>
    </div>
    <div
      className="display row"
    >
      <h3>
        application.values.contract
        :
      </h3>
      <div>
        <h4>
          no-connections
        </h4>
      </div>
    </div>
    <div
      className="display row"
    >
      <h3>
        application.values.application_dependency
        :
      </h3>
      <h4>
        no-connections
      </h4>
    </div>
    <div
      className="display-description row"
    >
      <h2>
        application.continuity
      </h2>
    </div>
    <div
      className="display row"
    >
      <div
        className="row"
      >
        <h3>
          application.values.update_practice
        </h3>
        <p>
          Update Practice
        </p>
      </div>
    </div>
    <div
      className="display row"
    >
      <div
        className="row"
      >
        <h3>
          application.values.security_practice_monitoring
        </h3>
        <p>
          Security Practice Monitoring
        </p>
      </div>
    </div>
    <div
      className="display row"
    >
      <div
        className="row"
      >
        <h3>
          application.values.recovery_practices_convalescence
        </h3>
        <p>
          Recovery Practices Convalescence
        </p>
      </div>
    </div>
    <div
      className="display row"
    >
      <div
        className="row"
      >
        <h3>
          application.values.user_rights_management
        </h3>
        <p>
          User Rights Management
        </p>
      </div>
    </div>
    <div
      className="display row"
    >
      <div
        className="row"
      >
        <h3>
          application.values.security_solutions
        </h3>
        <p>
          Security Solutions
        </p>
      </div>
    </div>
    <div
      className="display-description row"
    >
      <h2>
        application.responsibility
      </h2>
    </div>
    <div
      className="display row"
    >
      <div
        className="row"
      >
        <h3>
          application.values.product_owner
        </h3>
        <p>
          Product Owner
        </p>
      </div>
    </div>
    <div
      className="display row"
    >
      <div
        className="row"
      >
        <h3>
          application.values.application_holder
        </h3>
        <p>
          Application Holder
        </p>
      </div>
    </div>
    <div
      className="display row"
    >
      <div
        className="row"
      >
        <h3>
          application.values.admin_users
        </h3>
        <p>
          Admin Users
        </p>
      </div>
    </div>
    <div
      className="display row"
    >
      <div
        className="row"
      >
        <h3>
          application.values.liability_professional_users
        </h3>
        <p>
          Liability Professional Users
        </p>
      </div>
    </div>
    <div
      className="display row"
    >
      <div
        className="row"
      >
        <h3>
          application.values.holder_extra_info
        </h3>
        <p>
          Holder Extra Info
        </p>
      </div>
    </div>
    <div
      className="row"
    >
      <h3>
        Käyttöoikeusvastaavat:
      </h3>
      <p />
    </div>
    <div
      className="display row"
    >
      <h3>
        application.values.provider
        :
      </h3>
      <div>
        <h4>
          no-connections
        </h4>
      </div>
    </div>
    <div
      className="display row"
    >
      <div
        className="row"
      >
        <h3>
          application.values.provider_responsibility
        </h3>
        <p>
          Provider Responsibility
        </p>
      </div>
    </div>
    <div
      className="display row"
    >
      <div
        className="row"
      >
        <h3>
          application.values.additional_contacts
        </h3>
        <p>
          Additional Contacts
        </p>
      </div>
    </div>
    <div
      className="display-description row"
    >
      <h2>
        application.extra
      </h2>
    </div>
    <div
      className="display row"
    >
      <div
        className="row"
      >
        <h3>
          application.values.known_issues
        </h3>
        <p>
          Known Issues
        </p>
      </div>
    </div>
    <div
      className="action-buttons"
    >
      <button
        className="action-button btn btn-secondary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        button.copy
      </button>
      <button
        className="action-button btn btn-secondary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        button.delete
      </button>
      <button
        className="action-button btn btn-secondary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        button.edit
      </button>
    </div>
  </div>
</div>
`;
